syntax = "proto3";

package chime;

enum DataType {
  DT_INVALID = 0;
  DT_CHAR = 1;
  DT_FLOAT32 = 2;
  DT_FLOAT64 = 3;
  DT_INT32 = 4;
  DT_INT64 = 5;
  DT_HALF = 6;
  DT_FLOAT16 = 7;
  DT_BFLOAT16 = 8;
  DT_BOOL = 9;
  DT_INT8 = 10;
  DT_INT16 = 11;
  DT_INT128 = 12;
  DT_STRING = 13;
  DT_UINT8 = 14;
  DT_UINT16 = 15;
  DT_UINT32 = 16;
  DT_UINT64 = 17;
  DT_UINT128 = 18;
  DT_FLOAT128 = 19;
  DT_COMPLEX32 = 20;
  DT_COMPLEX64 = 21;
  DT_COMPLEX128 = 22;
}
